src/main/java/io/cdap/e2e/pages/actions/CdfLogActions.java

```diff
@@ -17,56 +17,91 @@
 package io.cdap.e2e.pages.actions;
 
 import io.cdap.e2e.pages.locators.CdfLogLocators;
+import io.cdap.e2e.utils.AssertionHelper;
+import io.cdap.e2e.utils.ConstantsUtil;
 import io.cdap.e2e.utils.ElementHelper;
 import io.cdap.e2e.utils.SeleniumHelper;
 
 /**
- * Represents CdfLogActions
+ * Represents Cdf Logs Page Actions
  */
 public class CdfLogActions {
-  public static CdfLogLocators cdfLogLocators;
 
   static {
-    cdfLogLocators = SeleniumHelper.getPropertiesLocators(CdfLogLocators.class);
+     SeleniumHelper.getPropertiesLocators(CdfLogLocators.class);
   }
 
-  public static void validateErrorPopupLog() throws InterruptedException {
-    ElementHelper.clickOnElement(cdfLogLocators.errorMessagePopup);
-  }
-
-  public static void validateErrorPopupLog(String error) throws InterruptedException {
-    ElementHelper.clickOnElement(cdfLogLocators.errorMessagePopup);
+  /**
+   * Verify if 'The preview of the pipeline has failed' error message popup is displayed
+   */
+  public static void validateErrorPopupLog() {
+    AssertionHelper.verifyElementDisplayed(CdfLogLocators.errorMessagePopup);
   }
 
+  /**
+   * Click on the close button(x) on the error message
+   */
   public static void dismissPopup() {
-    cdfLogLocators.dismissable.click();
+    ElementHelper.clickOnElement(CdfLogLocators.dismissable);
   }
 
+  /**
+   * Click on the Logs button on the header
+   */
   public static void goToLogs() {
-    cdfLogLocators.goToLogs.click();
+    ElementHelper.clickOnElement(CdfLogLocators.goToLogs);
   }
 
+  /**
+   * Verify if 'Pipeline ... failed' log is shown
+   */
   public static void validateFailed() {
-    cdfLogLocators.validateFailed.isDisplayed();
+    AssertionHelper.verifyElementDisplayed(CdfLogLocators.validateFailed);
   }
 
+  /**
+   * Verify if 'Pipeline ... failed' log is shown with 'ERROR' Level
+   */
   public static void validateCategoryError() {
```

Comment:
What is this method trying to achieve? Hard to understand from method name? What do you mean by `validateCategoryError` ?